### DISCUSSION TITLE
cons reduce

### DISCUSSION THREAD

**Discussion by DanielAjoy** [h3rald/min#166] [Bugs: Fixed]
I tried this:

> (5) () (cons) reduce

And was expecting this:

> (5 ())

But instead I got

> (!) <repl>(1,12) [cons]: field 'qVal' is not accessible for type 'MinValueObject' using 'kind = minInt'
>     <repl>(1,12) in symbol: cons
>     <repl>(1,20) in symbol: reduce

Is it that **reduce** is being overly specific in its definition? Or am I missing something?

I saw that reduce does work with strings

> $ ("a" "b" "c") "---" (suffix) reduce
>  "---abc"

even though the reference says it produces ints

> quot1 a quot2 ⇒ int

**Comment by DanielAjoy** [comment]
Nevermind about the first part I got this to work:

> $ (1 2 3 4 5) () (swap cons) reduce
>  (
>    5
>    4
>    3
>    2
>    1
>   )

So my experiment to determine if reduce only produced items, now satisfies my. I conclude that it doesn't, it can produce quotes too.

But the last part remains. This signature is wrong, or a simplification of what reduce can do.

> quot1 a quot2 ⇒ int

**Comment by h3rald** [comment]
You are right, the signature was wrong, updated.
